apps/src/lib/levelbuilder/lesson-editor/AddResourceDialog.jsx

```diff
@@ -18,33 +60,212 @@ const styles = {
 export default class AddResourceDialog extends Component {
```

Comment:
It looks like you repurposed the existing AddResourceDialog for this purpose. Its kinda tricky but this was actually serving a slightly different add functionality that we will want to keep around.

1 (what you are add) Adding a new resource or updating existing resource information
2 (what this component was doing) Picking from the list of resources connected to the lesson and adding the key for that resource to the markdown for that activity section.

So maybe we should call them
1) AddNewResourceDialog
2) AddResourceInMarkdownDialog

However I think we need both because they are serving different purposes

Reply:
Yeah that makes sense. I originally created a new dialog then found this one that looked very similar, so I repurposed it. It wasn't until I realized that there was a button in the activity cards that I figured out that there was a secondary purpose. 

Thoughts on what to do for this PR? I see a couple options:
1. Leave as is and follow up with a new component for AddResourceInMarkdownDialog
2. Rename AddNewResourceDialog to CreateResourseDialog or similar, restore the old AddResourceDialog, and update this PR

Reply:
What about a third option: Create a PR that goes in ahead of this PR that renames the existing component. Then you can keep AddResourceDialog as is in this PR.

Reply:
Oh good call! I'll do that

Reply:
https://github.com/code-dot-org/code-dot-org/pull/37371